KotlinFutures/app/src/main/java/com/example/kotlinfutures/model/common/FingerprintHelper.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.kotlinfutures.model.common
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import android.support.v4.hardware.fingerprint.FingerprintManagerCompat
+import android.support.v4.hardware.fingerprint.FingerprintManagerCompat.AuthenticationCallback
+import android.support.v4.os.CancellationSignal
+import com.example.kotlinfutures.view.Callback
+import java.security.KeyStore
+import java.security.KeyStoreException
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+
+
+class FingerprintHelper(
+    private val fingerprintManager: FingerprintManagerCompat,
+    private val callback: Callback,
+    private var cryptoObject: FingerprintManagerCompat.CryptoObject? = null,
+    private var keyStore: KeyStore? = null,
+    private var keyGenerator: KeyGenerator? = null,
+    private var cancellationSignal: CancellationSignal? = null,
+    private var selfCancelled: Boolean = false
+) : AuthenticationCallback() {
+
+    init {
+        try {
+            keyStore = KeyStore.getInstance(ANDROID_KEY_STORE)
+            keyGenerator = KeyGenerator
+                .getInstance(KeyProperties.KEY_ALGORITHM_AES,
+                    ANDROID_KEY_STORE
+                )
+        } catch (exc: KeyStoreException) {
+            exc.printStackTrace()
```

Comment:
It would be better to use Timber or Log or create some feedback in case of getting an exception and not just printing it to the stack trace.